services/repository/mongodb/service.go

```diff
@@ -0,0 +1,173 @@
+	if err != nil {
+		return nil, repository.NewUnknownErrorWithError("database is not accessible.", err)
+	}
+
+ 	//Todo : Do we need to have GetDbName in configuration service ? for now it is just hard coded
+	collection := client.Database("tenantdb").Collection("tenants")
+	insertResult, err := collection.InsertOne(context.TODO(), request.Tenant)
+
+	if err != nil {
+    	return nil, repository.NewUnknownErrorWithError("Insert tenant failed.", err)
+	}
+
+	return &repository.CreateTenantResponse{
+		TenantID: insertResult.InsertedID.(string),
+	}, nil
+}
+
+
+// UpdateTenant update an existing tenant
+// context: Optional The reference to the context
+// request: Mandatory. The request to update an existing tenant
+// Returns either the result of updateing an exiting tenant or error if something goes wrong.
+func (service *mongodbRepositoryService) UpdateTenant(
+	ctx context.Context,
+	request *repository.UpdateTenantRequest) (*repository.UpdateTenantResponse, error) {
+
+	clientOptions := options.Client().ApplyURI(service.dbConnectionString)
+	client, err := mongo.Connect(context.TODO(), clientOptions)		
+	if err != nil {
+		return nil, repository.NewUnknownErrorWithError("Could not connect to mongodb database.", err)
+	}
+	
+	//Todo : make sure if ping is necessary
+	err = client.Ping(context.TODO(), nil)
+	if err != nil {
+		return nil, repository.NewUnknownErrorWithError("database is not accessible.", err)
+	}
+	
+	 //Todo : Do we need to have GetDbName in configuration service ? for now it is just hard coded
+	collection := client.Database("tenantdb").Collection("tenants")
+	filter := bson.D{{"TenantID", request.TenantID}}
+
+	newTenant := bson.D{
+		{"$set", request.Tenant},
+	}
+	_, err = collection.UpdateOne(context.TODO(), filter, newTenant)
+	if err != nil {
+    	return nil, repository.NewUnknownErrorWithError("Update tenant failed.", err)
+	}
+
+	return &repository.UpdateTenantResponse{}, nil
+}
+
+
+// DeleteTenant delete an existing tenant
+// context: Optional The reference to the context
+// request: Mandatory. The request to delete an existing tenant
+// Returns either the result of deleting an exiting tenant or error if something goes wrong.
+func (service *mongodbRepositoryService) DeleteTenant(
+	ctx context.Context,
+	request *repository.DeleteTenantRequest) (*repository.DeleteTenantResponse, error) {
+
+	clientOptions := options.Client().ApplyURI(service.dbConnectionString)
+	client, err := mongo.Connect(context.TODO(), clientOptions)		
+	if err != nil {
+		return nil, repository.NewUnknownErrorWithError("Could not connect to mongodb database.", err)
+	}
+		
+	//Todo : make sure if ping is necessary
+	err = client.Ping(context.TODO(), nil)
+	if err != nil {
+		return nil, repository.NewUnknownErrorWithError("database is not accessible.", err)
+	}
+		
+	 //Todo : Do we need to have GetDbName in configuration service ? for now it is just hard coded
+	collection := client.Database("tenantdb").Collection("tenants")
```

Comment:
name of database and collection should either come from the configuration service or must be defined as constant on top of this file